src/features/explore/exploreSlice.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  likePost,
+  bookmarkPost,
+  deletePost,
+  editPost,
+  uploadPost,
+} from "../posts/PostsSlice";
+import { exploreInitialState } from "./explore.types";
+
+const initialState: exploreInitialState = {
+  posts: [],
+  fetchPostsStatus: "idle",
+};
+
+export const fetchPosts = createAsyncThunk<Post[]>(
+  "explore/fetchPosts",
+  async () => {
+    const fetchedPosts: Post[] = [];
+    const q = query(collection(db, "posts"));
+    const snapshots = await getDocs(q);
+    snapshots.forEach((post) => {
+      fetchedPosts.push({ ...post.data(), id: post.id } as Post);
+      console.log(fetchedPosts);
+    });
+    return fetchedPosts as Post[];
+  }
+);
+
+const exploreSlice = createSlice({
+  name: "explore",
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(
+      fetchPosts.fulfilled,
+      (state, action: PayloadAction<Post[]>) => {
+        console.log(action.payload);
+        state.posts = action.payload;
+        state.fetchPostsStatus = "succeded";
+      }
+    );
+    builder.addCase(fetchPosts.rejected, (state) => {
+      state.fetchPostsStatus = "failed";
+    });
+    builder.addCase(fetchPosts.pending, (state) => {
+      state.fetchPostsStatus = "pending";
+    });
+    builder.addCase(
+      likePost.fulfilled,
+      (state, action: PayloadAction<ActionPostReturnType>) => {
+        if (
+          action.payload.explore === false &&
+          !isPresent({ arr: state.posts, value: action.payload.post.id ?? "" })
+        )
+          return;
+        const postIndex = state.posts.findIndex((post) => {
+          return post.id === action.payload.post.id;
+        });
+        if (postIndex !== -1) {
+          state.posts[postIndex] = action.payload.post;
+        }
+      }
+    );
+    builder.addCase(
+      bookmarkPost.fulfilled,
+      (state, action: PayloadAction<ActionPostReturnType>) => {
+        if (
+          !action.payload.explore &&
+          !isPresent({ arr: state.posts, value: action.payload.post.id ?? "" })
+        ) {
+          return;
+        }
+        const postIndex = state.posts.findIndex((post) => {
+          return post.id === action.payload.post.id;
+        });
+        if (postIndex !== -1) {
+          state.posts[postIndex] = action.payload.post;
+        }
+      }
+    );
+    builder.addCase(
+      deletePost.fulfilled,
+      (state, action: PayloadAction<DeletePostParams>) => {
+        if (
+          !action.payload.explore &&
+          !isPresent({ arr: state.posts, value: action.payload.postId ?? "" })
+        ) {
+          return;
+        }
+        state.posts = state.posts.filter((post) => {
+          return post.id !== action.payload.postId;
+        });
+      }
+    );
+    builder.addCase(
+      editPost.fulfilled,
+      (state, action: PayloadAction<ActionPostReturnType>) => {
+        if (
+          !action.payload.explore &&
+          !isPresent({ arr: state.posts, value: action.payload.post.id ?? "" })
```

Comment:
clean code good work ! keep it @Priya-Patel31

Reply:
Thank you !! @t007rushi